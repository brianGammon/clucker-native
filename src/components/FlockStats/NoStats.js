/* @flow */
import React from 'react';
import { Container, Text, Content } from 'native-base';
import HelpMessage from '../HelpMessage';
import Header from '../Header';

const NoStats = () => (
  <Container>
    <Header title="Flock Stats" eggButton />
    <Content padder>
      <HelpMessage>
        <Text>
          No eggs have been logged yet. Get started by tapping the &#34;+&#34;
          button above.
        </Text>
      </HelpMessage>
    </Content>
  </Container>
);

export default NoStats;
