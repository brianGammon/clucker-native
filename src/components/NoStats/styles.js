import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  message: {
    fontSize: 17,
    fontStyle: 'italic',
  },
});
