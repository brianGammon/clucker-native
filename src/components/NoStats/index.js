import NoStats from './NoStats';

export default NoStats;
