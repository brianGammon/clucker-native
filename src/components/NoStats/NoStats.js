import React from 'react';
import { View, Text } from 'native-base';
import Line from '../Line';
import styles from './styles';

type Props = {
  message: string | null,
};

const NoStats = ({ message }: Props) => (
  <View style={styles.container}>
    <Line />
    <Text style={styles.message}>
      {message || 'No stats available for this period'}
    </Text>
  </View>
);

export default NoStats;
