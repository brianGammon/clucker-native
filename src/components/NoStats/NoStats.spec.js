import React from 'react';
import NoStats from '.';

describe('NoStats', () => {
  test('renders with default message', () => {
    const wrapper = shallow(<NoStats />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with message prop', () => {
    const wrapper = shallow(<NoStats message="Passed in message" />);
    expect(wrapper).toMatchSnapshot();
  });
});
