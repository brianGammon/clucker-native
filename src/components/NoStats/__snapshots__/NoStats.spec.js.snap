// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoStats renders with default message 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Line />
  <Styled(Text)
    style={
      Object {
        "fontSize": 17,
        "fontStyle": "italic",
      }
    }
  >
    No stats available for this period
  </Styled(Text)>
</Styled(ViewNB)>
`;

exports[`NoStats renders with message prop 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Line />
  <Styled(Text)
    style={
      Object {
        "fontSize": 17,
        "fontStyle": "italic",
      }
    }
  >
    Passed in message
  </Styled(Text)>
</Styled(ViewNB)>
`;
